scripts/modules/b2b-account/email-quote.js

```diff
@@ -0,0 +1,123 @@
+    "hyprlivecontext",
+    'modules/mozu-grid/mozugrid-view',
+    'modules/mozu-grid/mozugrid-pagedCollection',
+    "modules/models-quotes"],
+    function ($, api, _, Hypr, Backbone, HyprLiveContext, MozuGrid, MozuGridCollection,QuoteModels) {
+
+
+        var accountInfo = require.mozuData('user');
+
+        var maxEmailAllowed = 20;
+
+        var emailQuoteView = Backbone.MozuView.extend({
+            templateName: "modules/b2b-account/quotes/email-quote",
+
+            initialize: function () {
+                Backbone.MozuView.prototype.initialize.apply(this, arguments);
+            },
+            render: function (quoteId) {
+                var self = this;
+                self.model.set("quoteId", quoteId);
+                Backbone.MozuView.prototype.render.apply(this, arguments);
+                if (!accountInfo.isSalesRep) {
+                    $("#sendEmailText").val(accountInfo.email);
+                }
+                $(".modal-dialog").attr("id", "mz-email-modal");
+
+                $('[data-mz-action="handleSendemail"]').click(function (e) {
+                    e.preventDefault();
+                    var invalidEmails = [];
+                    var validEmails = [];
+                    var emails;
+                    if (!self.model.attributes.inputEmails && !accountInfo.isSalesRep) {
+                        emails = $("#sendEmailText").val();
+                    }
+                    else {
+                        emails = self.model.attributes.inputEmails;
+                        if(emails === undefined){
+                            emails='';
+                        }
+                    }
+
+                    var result = emails.split(",");
+                    for (var i = 0; i < result.length && i < maxEmailAllowed; i++) {
+                        if (!result[i].trim().match(Backbone.Validation.patterns.email)) {
+                            if (result[i].trim() !== "") {
+                                invalidEmails.push(result[i].replace('<', '&lt').replace('>', '&gt'));
+                            }
+                        } else {
+                            validEmails.push(result[i]);
+                        }
+                    }
+
+                    var quoteId = self.model.get("quoteId");
+                    if (invalidEmails.length === 0 && validEmails.length !== 0) {
+                        var quote = new QuoteModels.Quote({
+                            id: quoteId
+                        });
+                        return quote.apiModel.email(validEmails).then(function (res) {
+                            self.showMessageBar({ message: "Email has been sent successfully" }, false);
+                            setTimeout(function () {
+                                $('#closeModal').click();
+                            }, 4000);
+
+                        }, function (error) {
+                            if (error.message !== undefined) {
```

Comment:
you dont have to check it against undefined just do a null check like below -

if(error.message){
          //statments here
}

Reply:
done